[Blazor] Forms - Binding - Custom input components - InputBase recommendation (#33495)
dotnet/AspNetCore.Docs: aspnetcore/blazor/forms/binding.md

For custom input processing scenarios, the following subsections demonstrate custom input components:

* [Input component based on `InputBase<T>`](#input-component-based-on-inputbaset): The component inherits from <xref:Microsoft.AspNetCore.Components.Forms.InputBase%601>, which provides a base implementation for binding, callbacks, validation, and more. Components that inherit from <xref:Microsoft.AspNetCore.Components.Forms.InputBase%601> must be used in a Blazor form (<xref:Microsoft.AspNetCore.Components.Forms.EditForm>).

* [Input component with full developer control](#input-component-with-full-developer-control): The component takes full control of input processing. The component's code must manage binding, callbacks, and validation. The component can be used inside or outside of a Blazor form.

> **Recommendation**: Derive your custom input components from <xref:Microsoft.AspNetCore.Components.Forms.InputBase%601> unless specific requirements prevent you from doing so. The <xref:Microsoft.AspNetCore.Components.Forms.InputBase%601> class is actively maintained by the ASP.NET Core team, ensuring it stays up-to-date with the latest Blazor features and framework changes. By building on this foundation, you benefit from continuous improvements and optimizations, minimizing the effort needed to adapt your components to future Blazor versions.

Reviewer: Let's shorten this up a bit by removing the last sentence. It might be a good idea to add a section on this subject to fundamentals overview content, but it's also clear to most that this is always the case with custom code.

```suggestion
We recommend that you derive your custom input components from <xref:Microsoft.AspNetCore.Components.Forms.InputBase%601>, unless specific requirements prevent you from doing so. The <xref:Microsoft.AspNetCore.Components.Forms.InputBase%601> class is actively maintained by the ASP.NET Core team, ensuring it stays up-to-date with the latest Blazor features and framework changes.
```
